lib/main/monitoring/app_runner.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:async';
+
+import 'package:core/utils/app_logger.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:tmail_ui_user/main/monitoring/sentry/sentry_manager.dart';
+
+/// Runs the app inside a guarded zone with logger + Sentry enabled.
+Future<void> runAppWithMonitoring(Future<void> Function() runTmail) async {
+  // Handling Asynchronous Errors
+  await runZonedGuarded(() async {
+    WidgetsFlutterBinding.ensureInitialized();
+
+    // Handling Flutter UI and Build Errors
+    FlutterError.onError = (details) async {
+      logError('FlutterError: ${details.exception} | stack: ${details.stack}');
+      await SentryManager.instance.reportError(
+        details.exception,
+        details.stack,
+      );
+      FlutterError.presentError(details);
+    };
+
+    // Handling Uncaught and Platform-Specific Errors
+    PlatformDispatcher.instance.onError = (error, stack) {
+      logError('PlatformDispatcher: Error: $error | stack: $stack');
+      SentryManager.instance.reportError(error, stack);
+      return true; // Prevent app from crashing
+    };
```

Comment:
we registered `onError` for unknown problem here @chibenwa

Reply:
IMO error are abnormal software behaviours that requires immediate DEV attention.

Things like Network connectivity are warn at best. Even maybe info. Even API failures. Warn a best: it's out of the flutter code scope and do not depict a problem with the app itself -> WARN.

Given the wy I would organize the log levels any logged error would deserve to go to Sentry.

(but that's my 2 cents)

Reply:
I'm agreed with @chibenwa on this point, if we `AppLoger.error()`, then we should sent it to sentry. but maybe we've used too much `.error` right now?

Reply:
Updated, use logError to send capture error to sentry